### ISSUE TITLE
The Hot Gym tutorial / validation set

### ISSUE THREAD

**Issue by zygmuntz** [numenta/nupic-legacy#1433]
I have a question about the hot gym tutorial. It seems that the example code operates on a whole data set both when training and predicting. Would it make sense to split the data into training and test sets, run the code on training set only and produce predictions for the test set?

https://github.com/numenta/nupic/tree/master/examples/opf/clients/hotgym
